nextcloud26: init at 26.0.0

RaitoBezarius opened the pull request:
###### Description of changes

:tada: 
See this slide for relevant changelog: https://youtu.be/docYiLbGSxI?t=5058
https://nextcloud.com/blog/hub-4-pioneers-ethical-ai-integration-for-a-more-productive-and-collaborative-future/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->
closes https://github.com/NixOS/nixpkgs/pull/219929

RaitoBezarius committed:
nextcloud26: init at 26.0.0

RaitoBezarius committed:
nextcloud26Packages: init at various

RaitoBezarius committed:
nextcloud25Packages: update at various

RaitoBezarius committed:
nextcloud24Packages: update at various

RaitoBezarius committed:
nixos/web-apps/nextcloud: introduce nextcloud26

SuperSandro2000 commented on pkgs/servers/nextcloud/patches/v26/0001-Setup-remove-custom-dbuser-creation-behavior.patch:
```diff
@@ -0,0 +1,157 @@
+ 	 * @param IDBConnection $connection
+ 	 */
+ 	private function createSpecificUser($username, $connection): void {
+-		$rootUser = $this->dbUser;
+-		$rootPassword = $this->dbPassword;
```
so we can drop https://github.com/NixOS/nixpkgs/blob/nextcloud26/pkgs/servers/nextcloud/patches/v25/0001-Setup-remove-custom-dbuser-creation-behavior.patch#L37-L38 , too?

RaitoBezarius replied:
as a cleanup PR for v25 yes, but I wouldn't bother too much? you can do it :P

icewind1991 commented:
the "remove custom dbuser creation" patch isn't required anymore for 26, you can now disable that behavior by setting `NC_setup_create_db_user=false` during setup or add `"setup_create_db_user" => false` to config.php before setup.

RaitoBezarius committed:
nixos/tests/nextcloud: do not expect failure on NC ≥26 with respect to RC4 bug

RaitoBezarius committed:
nextcloud26, nixos/web-apps/nextcloud: drop the autocreate db user patch and wire the default in the NixOS module

RaitoBezarius commented:
> the "remove custom dbuser creation" patch isn't required anymore for 26, you can now disable that behavior by setting `NC_setup_create_db_user=false` during setup or add `"setup_create_db_user" => false` to config.php before setup.

Done.

RaitoBezarius commented on nixos/modules/services/web-apps/nextcloud.nix:
```diff
@@ -747,7 +748,8 @@ in {
             ''
           else if versionOlder stateVersion "22.11" then nextcloud24
-          else nextcloud25
+          else if versionOlder stateVersion "23.05" then nextcloud25
```
I feel like the logic is fine.

Ma27 replied:
Agreed.

RaitoBezarius commented:
@Ma27 do you think we should remove NC24?

dotlambda commented on nixos/tests/nextcloud/openssl-sse.nix:
```diff
@@ -71,12 +72,15 @@ in {
         nextcloudwithopenssl1.succeed("grep bye /var/lib/nextcloud/data/root/files/test-shared-file2")
         nextcloudwithopenssl1.succeed("${withRcloneEnv3} ${pkgs.rclone}/bin/rclone cat nextcloud:test-shared-file2 | grep bye")
 
+
```
Why the random newline?

RaitoBezarius replied:
sorri

GaetanLepage commented:
Shouldn't it be added here as well ?
https://github.com/NixOS/nixpkgs/blob/60c1d71f2ba4c80178ec84523c2ca0801522e0a6/nixos/modules/services/web-apps/nextcloud.nix#L207

Ma27 commented:
I'll try to take a look later today, but no guarantees because exam phase.

Ma27 commented:
> @Ma27 do you think we should remove NC24?

Let's await the final patch release. Reason is quite simple: we can automatically backport all updates on the next updates and we don't have to fix up the backport.

Btw, I'd resolve the review comments and drop the nextcloud23 references (this makes the module simpler on some occasions) and deploy the changes to my setup. Would it be OK if I push the changes after that to your branch? :)

Ma27 committed:
nextcloud: remove compat references to v23

This is not needed anymore because the version is EOL for almost a year
now and we don't even have the packages anymore, only the attributes for
compatibility for upgrades from older NixOS versions.

Ma27 committed:
nixos/nextcloud: minor fixups

* Apply review comments.
* Default to PHP 8.2 for v26.

RaitoBezarius commented:
> > @Ma27 do you think we should remove NC24?
> 
> Let's await the final patch release. Reason is quite simple: we can automatically backport all updates on the next updates and we don't have to fix up the backport.

Got it.

> Btw, I'd resolve the review comments and drop the nextcloud23 references (this makes the module simpler on some occasions) and deploy the changes to my setup. Would it be OK if I push the changes after that to your branch? :)

Please do so! It's running on my setup since the opening of this PR, no trouble so far.

Ma27 committed:
nixos/nextcloud: release notes

Ma27 commented:
FWIW I got bitten by https://github.com/nextcloud/user_saml/issues/709, but considering that we don't ship SAML by default and the workaround is trivial I don't consider this a blocker.

Ma27 reviewed (approved):
LGTM. @RaitoBezarius would merge soonish if that's OK :)

RaitoBezarius commented:
> @RaitoBezarius btw, why does [881efdb](https://github.com/NixOS/nixpkgs/commit/881efdb55e1480f0c0fef905874ce1d7c1c118a3) state that the stateVersion logic is broken? The change seems OK to me :)

Fixed, excessive amount of carefulness. :)
